File: src/main/java/com/parmet/squashlambdas/activity/Sport.java
package com.parmet.squashlambdas.activity;

import static com.google.common.base.CaseFormat.UPPER_CAMEL;
import static com.google.common.base.CaseFormat.UPPER_UNDERSCORE;

public enum Sport {
  SQUASH,
  HARDBALL,
  TENNIS,
  RACQUETS;

  private final String pretty = UPPER_UNDERSCORE.to(UPPER_CAMEL, name());

  @Override
  public String toString() {
    return pretty;
  }
}

File: src/main/kotlin/com/parmet/squashlambdas/activity/Sport.kt
package com.parmet.squashlambdas.activity

import com.parmet.squashlambdas.notify.SportSerializer
import kotlinx.serialization.Serializable

@Serializable(with = SportSerializer::class)
sealed class Sport {
    override fun toString() =
        this::class.java.simpleName

    object Squash : Sport()

    object Hardball : Sport()

    object Tennis : Sport()

    object Racquets : Sport()

    object Fitness : Sport()
}
